### PR TITLE
Random seed doc

### LINKED ISSUE #20569: Potential error in documentation of keras.random.uniform.
In the documentation of keras.ranom.uniform I find

    seed: A Python integer or instance of
        `keras.random.SeedGenerator`.
        Used to make the behavior of the initializer
        deterministic. Note that an initializer seeded with an integer
        or None (unseeded) will produce the same random values
        across multiple calls. To get different random values
        across multiple calls, use as seed an instance
        of `keras.random.SeedGenerator`.

For me this implies that 

`keras.random.unform(shape=(), seed=None) == keras.random.unform(shape=(), seed=None)`

Which however is not the case.

See the colab here [https://colab.research.google.com/drive/19ITFpOVGBoVzBGWCXvMHsgUulVLjHuvm?usp=sharing](url) to see that calling keras.random.unform(shape=(), seed=None) produces a sequence of random numbers.

Am I misunderstanding this or is the documentation just wrong?

### PR DESCRIPTION
**Pull request by roebel**
Following the discussion under #20569 this pull request contains a reformulation of the outdated doc concerning
the seed argument of all RNG functions. I also added a clarification to the SeedGenerator class documentation.

Resolves #20569

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the improvement
